src/pages/edge/adobe-journey-optimizer/code-based/tutorial.md

```diff
@@ -0,0 +1,114 @@
+
+<Tabs query="platform=ios&function=using-propositions"/>
+
+## Tracking interactions with code-based experiences
+
+Since the onus of rendering the code-based experience lies with the app developer, you must monitor the desired end user interactions and call the appropriate tracking APIs. To record an interaction with the code-based content, call the [track](../public-classes/proposition-item.md#track) API provided in the `PropositionItem` class. The below code shows two examples of tracking: when the content is displayed to the user and when the user clicks on the content. These examples are for illustrating how to call the track  API and not a recommendation on where it should be called. Please examine your app workflow to find the appropriate way to perform tracking.
+
+<TabsBlock orientation="horizontal" slots="heading, content" repeat="2"/>
+
+Android
+
+<Tabs query="platform=android&function=track"/>
+
+iOS
+
+<Tabs query="platform=ios&function=track"/>
+
+### Tracking items from embedded decisions
+
+When Adobe Journey Optimizer campaigns are created with embedded decisions, the server can respond with one or more items based on the number of items requested in the decision. To track these embedded items, the Adobe Journey Optimizer campaign UI provides `item._trackingToken` attributes. When authoring the campaign's content, you would need to embed the provided token as an HTML data-attribute in case of HTML content or JSON attribute in case on JSON content. The below example campaign shows content with embedded decision, where multiple paragraph tags containing image URLs can be returned.
+
+```html
+<html>
+   <body>
+      {{#each decisionPolicy.1234567.items as |item|}}
+      <ul id={{item._id}} data-item-token="{{item._trackingToken}}"><img src="{{item.heroimage1.sourceURL}}" alt="Winter Sale" width="80%" height="50%"></ul>
+      {{/each}}
+   </body>
+</html>
+```
+
+For the above example, the server can return a response with two decision items in the item data within a single `PropositionItem`
+
+```html
+<html>
+   <body>
+      <ul id="mockId1" data-item-token="mockToken1"><img src="https://image1.jpeg" alt="Winter Sale" width="80%" height="50%"></ul>
+      <ul id="mockId2" data-item-token="mockToken2"><img src="https://image2.jpeg" alt="Winter Sale" width="80%" height="50%"></ul>
+   </body>
+</html>
+```
+
+Since the embedded items are located inside a single PropositionItem's data, the app developer will need to extract the `data-item-token` when tracking the interactions with the item. The token, along with the other tracking information, will need to be passed to the [track](../public-classes/proposition-item.md#track) API provided by the Messaging SDK. If no tracking tokens are supplied, normal tracking events will be sent. If tracking tokens were used during authoring, it will be assumed that all the corresponding embedded items were displayed. The below code shows an example of calling the track  API with tokens.
```

Comment:
One more double space between "track" and "API" here too!

Reply:
Ooops.. Got most of it this time!